seagap/configs/main.py

```diff
@@ -0,0 +1,110 @@
+"""main.py
+
+The main configuration module containing base settings pydantic
+classes
+"""
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+import yaml
+from pydantic import BaseSettings, Field
+from pydantic.fields import ModelField
+
+from .solver import Solver
+
+CONFIG_FILE = "config.yaml"
+
+
+def yaml_config_settings_source(settings: BaseSettings) -> Dict[str, Any]:
+    """
+    Read config settings form a local yaml file where the software runs
+
+    Parameters
+    ----------
+    settings : pydantic.BaseSettings
+        The base settings class
+
+    Returns
+    -------
+    dict
+        The configuration dictionary based on inputs from the yaml
+        file
+    """
+    encoding = settings.__config__.env_file_encoding
+    config_path = Path(CONFIG_FILE)
+    if config_path.exists():
+        # Only load config.yaml when it exists
+        return yaml.safe_load(config_path.read_text(encoding))
+    return {}
+
+
+class BaseConfiguration(BaseSettings):
+    """Base configuration class"""
+
+    @classmethod
+    def add_fields(cls, **field_definitions: Any) -> None:
```

Comment:
When is this needed? And is that the only reason to subclass ```BaseSettings```?

Reply:
Great question! This will be used when I setup the plugin functionality in the future. The reason I subclass `BaseSettings` is simply readability purposes. I was trying to make the `Configuration` class view simply a listing of fields and validation so it's like a model, and separate out the custom methods and configs.